### ISSUE TITLE
How to dispatch @@redux-form/UPDATE_SYNC_ERRORS actions manually ?

### ISSUE THREAD

**Issue by boy-techy** [erikras/redux-form#3788]
I want to Update Syncerrors in redux form but msg is set from somewhere after validating values of DatePicker: -

`import React from 'react';
import DatePicker from 'react-datepicker';
import moment from 'moment';
import { validateDateFormat } from "../../services/validators";

const CustomDatePicker = ({input, meta: {touched, error, dispatch, form}, label, className }) => {

    const validateDate = value => {
        let isValid = validateDateFormat(value);
        debugger;
        if(typeof isValid !== "boolean") {
            dispatch({
                type: "@@redux-form/UPDATE_SYNC_ERRORS",
                meta: {form},
                payload: {
                    syncErrors: {
                        [input.name]: isValid
                    }
                }
            });
        }
        /* else if(value.length === 6 && !/[^0-9]/i.test(value)) {
            if(value.substr(0,2) < 12 && value.substr(2,2) < 30){
                value = value.substr(0,2)+"/"+value.substr(2,2)+value.substr(4,2);
                input.onChange(moment(value).valueOf());
            }
        }*/
        else {
            input.onChange(moment(value).valueOf());
        }
    };

    return (
      <div className={`form-group ${className}`}>
              <label>{label}</label>
              <DatePicker name={input.name}
                          dateFormat='MM/DD/YY'
                          popperClassName="containerClass"
                          className={(touched  && error) ? 'error-input' : ''}
                          onBlur={(event) => validateDate(event.target.value)}
                          onChange={(date) => input.onChange(moment(date).valueOf())}
                          selected={input.value ? moment(input.value) : null}
                          placeholderText="MM/DD/YY" />
              <span className={`${ (touched  && error) && 'hidden' } icon-calendar show-calendar`} />
              <span className={`icon-alert error-icon ${ (touched  && error) && 'visible' } `} />
              <span className={`error error-msg ${ (touched  && error) && 'visible' } `}>{error}</span>
      </div>
  );
};

export default CustomDatePicker;`

But after dispatching "@@redux-form/UPDATE_SYNC_ERRORS" action in syncerror all errors are cleared out. 

So if there is any solution without connect to redux-store. Is there any option to collect all syncErrors of current form.

**Comment by gustavohenke** [comment]
Duplicate of #3516

**Comment by boy-techy** [comment]
SubmissionError also can't help me out, because I have to set the error before submission. I also approached dispatching @@redux-form/UPDATE_SYNC_ERRORS action with the available synErrors object and set a new one. But whenever I changed the value in above fields like name and others, I think validators reset the error of my custom error that is forcefully been set. So can anyone explain me from where payload is generating for syncErrors in the reduxForm.
